docs/architecture/maui/mvvm-community-toolkit-features.md

```diff
@@ -0,0 +1,211 @@
+> [!NOTE]
+> Even though the.NET MAUI `Command` and `RelayCommand` are similar, using `RelayCommand` allows for decoupling your ViewModel from any direct .NET MAUI references. This means that your ViewModel is more portable which can lead to easier reuse across project.
+
+`AsyncRelayCommand` provides many additional features when working with asynchronous workflows. This is quite common in our ViewModel as we are typically communicating with repositories, APIs, databases, and other systems that utilize `async/await`. The `AsyncRelayCommand` constructor takes in an execution task defined as a `Func<Task>` or a delegate returning `Task` as part of the constructor. While the execution task is running, `AsyncRelayCommand` will monitor the state of the task and provides updates using the `IsRunning` property. The `IsRunning` property can be bound to the UI which helps manage control states such as showing loading with an `ActivityIndicator` or disabling/enabling a control. While the execution task is being executed, the `Cancel` method can be called to attempt cancellation of the execution task, if supported.
+
+By default, `AsyncRelayCommand` does not allow concurrent execution. This is very helpful in situations where a user could unintentionally tap a control multiple times to execute a long-running or costly operation. During task execution, `AsyncRelayCommand` will automatically call the `CanExecuteChanged` event. In .NET MAUI, controls that support the `Command` and `CommandParameter` properties, such as `Button`, will listen to this event and automatically enable or disable it during execution. This functionality can be overridden by using a custom `canExecute` parameter or setting the `AsyncRelayCommandOptions.AllowConcurrentExecutions` flag in the constructor.
+
+For more detailed information on implementing commands, see the section [Implementing commands](mvvm.md#implementing-commands) in the MVVM chapter. Detailed information for the `RelayCommand` and `AsyncRelayCommand` is available in the [Commanding](/dotnet/communitytoolkit/mvvm/relaycommand) of the MVVM Toolkit Developer Center.
+
+## Source Generators
+
+Using the MVVM Toolkit components out-of-the-box allows us to greatly simplify our ViewModel. The MVVM Toolkit allows us to simplify common code use cases even further by using [Source Generators](/dotnet/csharp/roslyn-sdk/source-generators-overview). The MVVM Toolkit source generators look for specific attributes in our code and can generate wrappers for properties and commands.
+
+> [!NOTE]
+> The MVVM Toolkit Source Generators generate code that is additive to our existing objects. Because of this, any object that is leveraging a source generator will need to be marked as `partial`.
+
+The MVVM Toolkit `ObservableProperty` attribute works on classes based-off on `ObservableObject` and will wrap a private variable as a property that generates changes. The following code shows an example of using the `ObservableObject` attribute:
```

Comment:
I'm not sure how to change this, but this part of the sentence

```md
works on classes based-off on
```

doesn't make sense to me. Could you please rephrase that?

Reply:
Yea, it reads weird for sure. On it now.

Reply:
I updated it, and I think it is better, but it could be more concise. I wasn't sure of a better way to make sure that it was explicit enough to show where it can be used.